aws/data_source_aws_rds_orderable_db_instance_test.go

```diff
@@ -0,0 +1,167 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccAWSRdsOrderableDbInstanceDataSource_basic(t *testing.T) {
+	dataSourceName := "data.aws_rds_orderable_db_instance.test"
+	engine := "mysql"
+	engineVersion := "5.7.22"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSRdsOrderableDbInstance(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSRdsOrderableDbInstanceDataSourceConfigBasic(engine, engineVersion),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "engine", engine),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_version", engineVersion),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSRdsOrderableDbInstanceDataSource_preferred(t *testing.T) {
+	dataSourceName := "data.aws_rds_orderable_db_instance.test"
+	engine := "mysql"
+	engineVersion := "5.7.22"
+	preferredOption := "db.t2.small"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSRdsOrderableDbInstance(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSRdsOrderableDbInstanceDataSourceConfigPreferred(engine, engineVersion, preferredOption),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "engine", engine),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_version", engineVersion),
+					resource.TestCheckResourceAttr(dataSourceName, "db_instance_class", preferredOption),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSRdsOrderableDbInstanceDataSource_versionPrefix(t *testing.T) {
+	dataSourceName := "data.aws_rds_orderable_db_instance.test"
+	engine := "mysql"
+	engineVersionPrefix := "5.7"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSRdsOrderableDbInstance(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSRdsOrderableDbInstanceDataSourceConfigVersionPrefix(engine, engineVersionPrefix),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "engine", engine),
+					resource.TestCheckResourceAttr(dataSourceName, "engine_version", "5.7.30"),
```

Comment:
I would be concerned that this value may update regularly over time and that we might want to fetch the true value in a PreCheck, but I have additional comments about this functionality that I'll write up separately.